### ISSUE TITLE
Loading best model yields different prediction results

### ISSUE THREAD

**Issue by Mokers1234** [ThilinaRajapakse/simpletransformers#813]
Hello,

I split my data into train and validation sets and then train as below, saving best model:

```
args = {
                    'reprocess_input_data': True,
                    'overwrite_output_dir': True,
                    'manual_seed': 42,
                    'output_dir': self.bert_folder,
                    'best_model_dir': self.bert_folder + 'best_model/',
                    'evaluate_during_training': True
                    }
                self.vectorizer = ClassificationModel('distilbert',
                        'distilbert-base-uncased',
                        num_labels=len(self.train_df['DECILE_POSITION'
                        ].unique()), args=args)

                self.df_x = self.train_df[['COMMENT', 'DECILE_POSITION'
                        ]]
                self.df_x.rename(columns={'COMMENT': 'text',
                                 'DECILE_POSITION': 'labels'},
                                 inplace=True)
                
                split = int(len(self.df_x) * 0.8)
                self.train = self.df_x[:split]
                self.val = self.df_x[split:]
                self.vectorizer.train_model(self.train,
                        eval_df=self.val)
```
This yields 23% prediction accuracy on my test set (mine is a very noisy task).  Upon then loading the best model as below:

```
args = {
                    'reprocess_input_data': True,
                    'manual_seed': 42
                    }
                self.vectorizer = ClassificationModel('bert',
                        self.bert_folder + 'best_model',
                        num_labels=len(self.train_df['DECILE_POSITION'].unique()),
                        args=args)
```
The same accuracy function on the same test set produces only a 9% accuracy. Is there something I'm missing? The model, once loaded, should theoretically produce the same prediction results.

**Comment by ThilinaRajapakse** [comment]
By default, the model with the lowest `eval_loss` is taken as the best model. Could it be possible that the model with the lowest `eval_loss` doesn't have the best accuracy?

You can change this behaviour by using the `early_stopping_metric` and `early_stopping_metric_minimize` model arguments.

E.g.:

```python
model_args = ClassificationArgs()
model_args.early_stopping_metric_minimize = False
model_args.early_stopping_metric_metric = "accuracy"
```

**Comment by AlexMikhalev** [comment]
Bump. I suggest putting it into docs - it's really.

**Comment by ninasujit2016** [comment]
I have the same problem. 
I tried to use early_stopping_metric and early stopping metric_minimize model arguments, but still when i reload the model, the predciton on the same data is not so good as the one during the training process. Can you tell me where I may got wrong ?
